cxx-sensors/src/tools/cppcheck_createrules.py

```diff
@@ -48,88 +52,177 @@ def indent(elem, level=0):
         if level and (not elem.tail or not elem.tail.strip()):
             elem.tail = i
 
-# monkey pathing element tree to support CDATA as by 
-# Eli Golovinsky, 2008, www.gooli.org
-#            
+# see https://stackoverflow.com/questions/174890/how-to-output-cdata-using-elementtree
 def CDATA(text=None):
-    """
-    A CDATA element factory function that uses the function itself as the tag
-    (based on the Comment factory function in the ElementTree implementation).
-    """
-    element = et.Element(CDATA)
+    element = et.Element('![CDATA[')
     element.text = text
     return element
 
-old_ElementTree = et.ElementTree
-class ElementTree_CDATA(old_ElementTree):
-    def _write(self, file, node, encoding, namespaces):
-        if node.tag is CDATA:
-            if node.text:
-                text = node.text.encode(encoding)
-                file.write("<![CDATA[%s]]>" % text)
-        else:
-            old_ElementTree._write(self, file, node, encoding, namespaces)
-et.ElementTree = ElementTree_CDATA
-       
+et._original_serialize_xml = et._serialize_xml
+
+def _serialize_xml(write, elem, encoding, qnames, namespaces):
+    if elem.tag == '![CDATA[':
+        write("<%s%s]]>%s" % (elem.tag, elem.text, "" if elem.tail is None else elem.tail))
+    else:
+        et._original_serialize_xml(write, elem, encoding, qnames, namespaces)
+
+et._serialize_xml = et._serialize['xml'] = _serialize_xml
+
+
 def header():
-    return '<?xml version="1.0" encoding="ASCII"?>\n'
+    return '<?xml version="1.0" encoding="us-ascii"?>\n'
```

Comment:
@ivangalkin why not utf-8?

Reply:
@guwirth this is the same encoding, which is used in cxx-sensors/src/main/resources/cppcheck.xml
Please see https://github.com/ivangalkin/sonar-cxx/blob/a920b0c2c2ef67526a69c7d374b659981135725c/cxx-sensors/src/main/resources/cppcheck.xml
I don't know why it was selected in the first place.

Reply:
@ivangalkin looking to the cppcheck report it's UTF-8. Think it's better to change it also: https://github.com/SonarOpenCommunity/sonar-cxx/blob/master/integration-tests/testdata/cppcheck_project/cppcheck-v2.xml